pandas/tests/formats/test_printing.py

```diff
@@ -121,6 +122,61 @@ def test_ambiguous_width(self):
         self.assertEqual(adjoined, expected)
 
 
+class TestTableSchemaRepr(tm.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        tm._skip_if_no_mock()
+        tm._skip_if_no_ipython()
+        try:
+            import mock
+        except ImportError:
+            from unittest import mock
+        cls.mock = mock
+
+    def test_publishes(self):
+        df = pd.DataFrame({"A": [1, 2]})
+        objects = [df['A'], df, df]  # dataframe / series
+        expected_keys = [
+            {'text/plain', 'application/vnd.dataresource+json'},
+            {'text/plain', 'text/html', 'application/vnd.dataresource+json'},
+        ]
+
+        make_patch = self.mock.patch('IPython.display.display')
+        for obj, expected in zip(objects, expected_keys):
+            with make_patch as mock_display:
+                handle = obj._ipython_display_()
+                self.assertEqual(mock_display.call_count, 1)
+                self.assertIsNone(handle)
+                args, kwargs = mock_display.call_args
+                arg, = args  # just one argument
+
+            self.assertEqual(kwargs, {"raw": True})
+            self.assertEqual(set(arg.keys()), expected)
+
+        with_latex = pd.option_context('display.latex.repr', True)
+
+        with with_latex, make_patch as mock_display:
+            handle = obj._ipython_display_()
+            args, kwargs = mock_display.call_args
+            arg, = args
+
+        expected = {'text/plain', 'text/html', 'text/latex',
+                    'application/vnd.dataresource+json'}
+        self.assertEqual(set(arg.keys()), expected)
+
+    def test_config_on(self):
+        df = pd.DataFrame({"A": [1, 2]})
+        result = df._repr_table_schema_()
+        self.assertIsNotNone(result)
+
+    def test_config_off(self):
+        df = pd.DataFrame({"A": [1, 2]})
+        with pd.option_context("display.html.table_schema", False):
```

Comment:
I'll double check, but the option is defined under the `display` context manager, so the correct usage. Or earlier, would you saying that this should be under a top-level `html` namespace?